src/vs/workbench/contrib/mcp/browser/mcpResourceQuickAccess.ts

```diff
@@ -391,21 +395,45 @@ export abstract class AbstractMcpResourceAccessPick {
 		}));
 
 		store.add(picker.onDidAccept(async event => {
-			if (!event.inBackground) {
-				picker.hide(); // hide picker unless we accept in background
-			}
-
-			if (runOptions?.handleAccept) {
-				runOptions.handleAccept?.(picker.activeItems[0], event.inBackground);
-			} else {
-				const [item] = picker.selectedItems;
-				const uri = await helper.toURI((item as ResourceQuickPickItem).resource);
-				if (uri) {
-					this._editorService.openEditor({ resource: uri, options: { preserveFocus: event.inBackground } });
+			try {
+				picker.busy = true;
+				if (runOptions?.handleAccept) {
+					runOptions.handleAccept?.(picker.activeItems[0], event.inBackground);
```

Comment:
We should handle this case too -- `handleAccept` is used in the "Attach Context" flow from the 📎 button in chat, and we want to ensure those are only 'files' as well